libs/core/version/include/hpx/version.hpp

```diff
@@ -17,56 +17,56 @@
 namespace hpx {
 
     // Returns the major HPX version.
-    [[nodiscard]] HPX_CORE_EXPORT std::uint8_t major_version();
+    HPX_MODULE_EXTERN_CORE [[nodiscard]] std::uint8_t major_version();
```

Comment:
I think this has to be in a reverse sequence, it may generate compilation errors otherwise.
```suggestion
    [[nodiscard]] HPX_MODULE_EXTERN_CORE std::uint8_t major_version();
```

Reply:
> Thanks for pointing this out. I tried your suggested ordering, but it led to compilation errors. For now, I had to keep the current sequence to avoid those errors.

Clangcl fails compiling if you have it the other way around (e.g., https://github.com/STEllAR-GROUP/hpx/actions/runs/16845463262/job/47736319598). So it looks like that for the `[[nodiscard]]` functions we need to split things up.

Reply:
Sorry about the confusion. I tried your suggested ordering again and it worked. I think the previous compilation errors I saw were related to something else.